syntax = "proto3";


service DataSetService {
    rpc createIterator(DatSetSpec) returns (IteratorSpec) {}
    rpc getNextBatch(IteratorSpec) returns (Dataset) {}
    rpc putDatapoint(KVPair) returns (KVPair) {}
    rpc getDatapoint(Key) returns (Value) {}
}


message Key {
    string k = 1;
}

message Value {
    string v = 1;
}

message KVPair{
    Key k = 1;
    Value v = 2;
}

message DatSetSpec {
    int32 Id = 1;
}

message IteratorSpec {
    int32 Id = 1;
}

message DataSetIteratorConfig {
    int32 batchSize = 1;
    bool shuffle = 2;
    int32 shuffleSeed = 3;
}

message Dataset {
    string data = 1;
    string schema = 2;
}
